practice/blackjack/main.py

```diff
@@ -0,0 +1,120 @@
+"""Blackjack playing game."""
+
+from hand import *
+from deck import *
+from player import *
+from card import *
+
+
+def play_round(card_deck, plyr):
+    """Deal the cards to whomever wants them.
+
+    Dealer:  automatically hits dealer until score is over 17
+    Non-dealer:  hits until they stay or score is over 21
+    """
+    if plyr.score == 21:
+        return
+    not_done = True
+    while not_done:
+        if plyr.name == 'Dealer':
+            if plyr.score >= 17:
+                return
+        else:
+            display_player_hand(plyr)
+            prompt_string = '{} hit or stay (h/s)? '.format(plyr.name)
+            response = input(prompt_string)
+            if response[0].lower() == 's':
+                return
+        card = draw_card(card_deck)
+        plyr.hand = add_card_to_hand(card, plyr.hand)
+        plyr.score = return_score_of_hand(plyr.hand)
+        if plyr.score > 21:
+            print('{} busted!'.format(plyr.name))
+            not_done = False
+    return
+
+
+def display_who_won(players):
+    """Displays who won at the end of the game.
+
+    >>> display_who_won([Player('Eric', Hand([]), 17),
+    ...                  Player('Dealer', Hand([]), 18)])
+    Dealer wins this hand.
+    >>> display_who_won([Player('Eric', Hand([]), 17),
+    ...                  Player('Dealer', Hand([]), 22)])
+    Eric wins this hand.
+    >>> display_who_won([Player('Eric', Hand([]), 17),
+    ...                  Player('Dealer', Hand([]), 17)])
+    Push.
+    >>> display_who_won([Player('Eric', Hand([]), 22),
+    ...                  Player('Dealer', Hand([]), 22)])
+    Nobody wins.
+    """
+    winning_point_amt = max([p.score if p.score <= 21 else 0 for p in players])
+    winners = [p.name for p in players if p.score == winning_point_amt]
```

Comment:
I would separate the winner calculation code from the output code. Then you can test the winner calculation code more exactly.